Create program for ADADAS dataset

kaz462 opened the pull request:
adadas initial draft, pending updates in define.xml

SHAESEN2 commented on vignettes/adam-adadas.Rmd:
```diff
@@ -0,0 +1,155 @@
+---
```
Why do we need an RMD file for ADaM creation?

thomas-neitmann replied:
No, we discussed on Friday to use plain `.R` files.

SHAESEN2 commented on vignettes/ectd/adam-adadas.R:
```diff
@@ -0,0 +1,133 @@
+## ---- message = FALSE-----------------------------------------------------------------------------------------------
```
name of file should be adadas.R, not adam-adadas.R.

SHAESEN2 commented on vignettes/ectd/adam-adadas.R:
```diff
@@ -0,0 +1,133 @@
+source(file.path(rprojroot::find_root("DESCRIPTION"), "inst/startup.R"))
+
+## -------------------------------------------------------------------------------------------------------------------
+# Working directory requires write permission
```
We should agree on a standard workflow at the top of the program. Typically you dont need to initialize the startup.R and do a check for the writing permission.

I only have two lines:

# Set up ------------------------------------------------------------------

fcts <- c("eff_models.R", "fmt.R", "helpers.R", "Tplyr_helpers.R")
invisible(sapply(fcts, FUN = function(x) source(file.path("R/", x), )))

thomas-neitmann replied:
Agree, no need for sourcing the startup script and checking write permissions. I think we don't even need the two lines you suggested @SHAESEN2. Since the project is a R package and the functions are in the `R/` folder we just have to `library()` the pakage to make the function available.

SHAESEN2 replied:
> Thanks @SHAESEN2 @thomas-neitmann for the review! I've addressed the above comments, also added program headers as @SHAESEN2 did in `submission/programs/adtte.R`.
> 
> Please suggest on folder structures for `.R` and output xpt dataset. I put `adadas.R` and `qc_adadas.R `under vignettes/ectd/ for now as discussed in the last meeting.

@kaz462 I proposed a folder structure for the dataset and program in #21 #27 waiting for team approval. Not sure where the qc program should live. Perhaps a qc folder in root?

SHAESEN2 commented on vignettes/ectd/adam-adadas.R:
```diff
@@ -0,0 +1,133 @@
+
+
+## -------------------------------------------------------------------------------------------------------------------
+## placeholder for using metacore/metatools
```
Missing:
1. QC versus the original xpt file to ensure you have 100% match
2. export dataset. I propose haven::write_xpt(adtte, file.path("submission/datasets/adtte.xpt")) to mimick an actual submission environment

SHAESEN2 replied:
Let me know if you need any help :)

thomas-neitmann replied:
For 2. please use {xportr} which works in conjunction woth {metacore} and {metatools} to create a valis `.xpt` file.

thomas-neitmann replied:
For 1. I would propose to put the comparing of datasets in a separate file `qc_adas.R` which reads in the original SAS dataset, the dataset you created and uses `diffdf::diffdf()` for comparison.

thomas-neitmann commented on vignettes/adam-adadas.Rmd:
```diff
@@ -0,0 +1,155 @@
+  dataset_add = aw_lookup,
+  by_vars = vars(AVISIT)
+) %>%
+  mutate(
+    AWTDIFF = abs(AWTARGET - ADY),
+    AWU = "DAYS"
+  )
```
A bit overkill but please use `derive_vars_duration()`.

kaz462 replied:
`derive_vars_duration()` is not applicable here:
* `derive_vars_duration()`  derives duration from **dates**, but AWTARGET is the target day within the visit window defined in SAP, not dates;
* AWTDIFF is the absolute difference between AWTARGET - ADY, not duration.

thomas-neitmann replied:
I see, never mind then 🙂

kaz462 commented:
Thanks @SHAESEN2 @thomas-neitmann for the review! I've addressed the above comments, also added program headers as @SHAESEN2 did in `submission/programs/adtte.R`. 

Please suggest on folder structures for `.R` and output xpt dataset. I put `adadas.R` and `qc_adadas.R `under vignettes/ectd/ for now as discussed in the last meeting.

SHAESEN2 commented:
> Thanks @SHAESEN2 @thomas-neitmann for the review! I've addressed the above comments, also added program headers as @SHAESEN2 did in `submission/programs/adtte.R`.
> 
> Please suggest on folder structures for `.R` and output xpt dataset. I put `adadas.R` and `qc_adadas.R `under vignettes/ectd/ for now as discussed in the last meeting.

@kaz462 I proposed a folder structure for the dataset and program in #21 #27 waiting for team approval. Not sure where the qc program should live. Perhaps a qc folder in root?

thomas-neitmann commented on vignettes/ectd/qc_adadas.R:
```diff
@@ -0,0 +1,17 @@
+library(diffdf)
+library(dplyr)
+
+# read source -------------------------------------------------------------
+#qc_adadas <- haven::read_xpt(file.path("adam", "adadas.xpt"))
+
+# compare R generated adadas with adam/adadas -----------------------------
+#diffdf(adadas , qc_adadas , suppress_warnings = T)
```
Why is this commented? Also keys have to be defined for the `diffdf()` call.

thomas-neitmann commented on vignettes/ectd/adadas.R:
```diff
@@ -0,0 +1,141 @@
+adsl <- haven::read_xpt(file.path("adam", "adsl.xpt"))
+
+
+## -------------------------------------------------------------------------------------------------------------------
```
Please remove these kind of comment lines, they don't serve a purpose.

thomas-neitmann commented:
I like the structure of `submission/datasets` and `submission/programs`. As the QC program will not be part of the submission a `qc` folder in the root directly makes sense. Please remember to add these non.standard directories (for an R package) to `.Rbuildignore`. Thanks!

thomas-neitmann commented on renv.lock:
```diff
@@ -1409,7 +1484,7 @@
       "Version": "0.16.0",
       "Source": "Repository",
-      "Repository": "CRAN",
+      "Repository": "RSPM",
```
```suggestion
      "Repository": "CRAN",
```

thomas-neitmann commented:
I'm going ahead and merge this even though one parameter has still discrepancies as I'd like to get some updates made on this branch onto `main`. @kaz462 Please open a new issue and PR for the final updates once you are back. Thanks!
